src/pages/home/report/ReportActionsList.js

```diff
@@ -145,6 +141,10 @@ class ReportActionsList extends React.Component {
     }
 
     render() {
+        // Non-animating skeleton UI represents that chat history exists but is not loaded locally in Onyx.
+        // While offline if there is at least one non-pending action or if this is a new chat then we don't need to show the non-animating skeleton UI.
+        const shouldShowNonAnimatingSkeleton = !this.props.report.isOptimisticReport && _.every(this.props.sortedReportActions, action => action.pendingAction);
```

Comment:
```suggestion
        const shouldShowStaticSkeleton = !this.props.report.isOptimisticReport && _.every(this.props.sortedReportActions, action => action.pendingAction);
```

We generally prefer variables that describe what something **is** vs. what it **is not**.

Reply:
Alternatively, change this to `shouldAnimateSkeleton` and then we don't have to invert later `!shouldShowNonAnimatingSkeleton`. I think I like that suggestion better.

Reply:
Seems unrelated to the changes here. But I'm kind of confused about why and how `isOptimisticReport` got introduced. I'd think that an optimistic `report` would have a `pendingAction` and not a separate flag to denote it as "optimistic".

Reply:
We have decided to show the static skeleton UI for the chats that we know exist on the server but aren't loaded locally in ONYX. 

Now if the chat is not loaded locally and we are offline then we can still take optimistic action but we will need to keep showing the static skeleton UI. But in the case of a new chat report, we don't have to show the static skeleton UI because we know this chat report doesn't exist on our server. But if I just check by `pendingAction` then static skeleton UI would show up for new chats as well so to filter them out I have added the condition to see if it is an optimistic report(which I think denotes a new chat created while offline).

Reply:
Aaah, my bad. I see your question now. But to be honest I don't have any explanation for this one(will have to check the blame). I used it because it was right in the report object but this could have been easily computed as well.